governance/xc_admin/packages/xc_admin_frontend/components/ProgramSwitch.tsx

```diff
@@ -0,0 +1,101 @@
+import { ProgramType, PROGRAM_TYPE_NAMES } from '@pythnetwork/xc-admin-common'
+import { useProgramContext } from '../contexts/ProgramContext'
+import { Menu, Transition } from '@headlessui/react'
+import { Fragment } from 'react'
+
+const Arrow = ({ className }: { className?: string }) => (
+  <svg
+    className={className}
+    width="10"
+    height="6"
+    viewBox="0 0 10 6"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M1 1L5 5L9 1"
+      stroke="currentColor"
+      strokeWidth="1.5"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    />
+  </svg>
+)
+
+/**
+ * Format display name from enum name (e.g., "PYTH_CORE" -> "Pyth Core")
+ */
+const formatDisplayName = (value: ProgramType): string => {
+  const enumName = PROGRAM_TYPE_NAMES[value]
+  return enumName
+}
+
+/**
+ * Component that allows users to switch between different Pyth programs
+ * (Core, Lazer, etc.)
+ */
+const ProgramSwitch = ({ light = false }: { light?: boolean }) => {
+  const { programType, setProgramType } = useProgramContext()
+
+  // Convert enum to array of options
+  const programOptions = Object.values(ProgramType)
+    .filter((value): value is ProgramType => typeof value === 'number') // Filter out reverse mappings
+    .map((value) => ({
+      value,
+      label: formatDisplayName(value),
+    }))
```

Comment:
I think this can just be `Object.entries(PROGRAM_TYPE_NAMES).map(([value, label]) => ({ value, label }))`